### PR TITLE
Add command to view templating config

### PR DESCRIPTION
**Pull request by bobwhitelock**
This PR adds a new command, `metal view-config` which shows the configured Metalware templating config, optionally in the context for a particular node. This shows as JSON the exact config which will be used when rendering templates in general (as used in `dhcp`, `hosts`, and `render` without a node passed) or for a particular node (as used in `build` and `render` with a node passed).

This replaces and supersedes the previous `show_config` script I threw together before, with a few improvements:

- this is an actual Metalware command, so it is always available rather than needing to copy scripts around;

- the full templating config is now shown, including the `alces` namespace and optionally the `alces.groups` namespaces and nodes within these;

- syntax highlighting of the output JSON is provided via [jq](https://stedolan.github.io/jq/) - additionally since `jq` is now installed along with Metalware, this also means you can pipe the output of this command to it as needed to further filter the output to get the parts of the config you want, e.g. the following command just shows the `alces` namespace:

```
   [root@controller(dev) metalware]# bin/metal view-config node01 | jq .alces
{
  "answers": {
    "cluster_name": "foo",
    "domain": "<%= alces.answers.cluster_name %>.cluster.local",
    "root_password": "password from domain",
[ ... etc ... ]
```

See the [description](https://github.com/alces-software/metalware/compare/develop...feature/viewing-templating-config#diff-3d8becb4f43e68f36881f3f383d363a7R307) for more details of what the command shows and some limitations.

@ColonelPanicks: I've added you as a reviewer in case you want to have a look before this is merged, and see how this fulfils your use cases for checking the config when working with Metalware. If you do you may want to install Metalware from the next PR instead, as that will be based on this one but with additional commands for viewing the answers.

### CODE REVIEW

File: src/commands/view_config.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        # Delegate pretty printing with colours to `jq`.
+        Open3.popen2(jq_command) do |stdin, stdout|
+          stdin.write(json)
+          stdin.close
+          puts stdout.read
+        end
+      end
+
+      def jq_command
+        "jq . #{colourize_output? ? '--color-output' : ''}"
```
Comment (WilliamMcCumstie) [inline src/commands/view_config.rb]: Does this need to be a command, or can it be a constant?

Comment (bobwhitelock) [inline src/commands/view_config.rb]: What do you mean by 'a command' - do you mean a function? If so it does as whether we colour the output depends on if we've been passed the option to do so.

Comment (WilliamMcCumstie) [inline src/commands/view_config.rb]: So I meant a method. Could we not use:
```
JQ_Command = "jq . #{colourize_output? ? '--color-output' : ''}".freeze
```

Comment (WilliamMcCumstie) [inline src/commands/view_config.rb]: Ohh sorry my bad, missed the variable substitution. It's good as is

Comment (bobwhitelock) [inline src/commands/view_config.rb]: OK - thanks for reviewing :slightly_smiling_face:

**Comment by ColonelPanicks** [comment]
Apologies, I missed the notification for this. 

Just tried to setup a local copy and `jq` isn't found so the `runtime` stage of the installer fails. This package is in the `epel` repos so `yum install epel-release` fixes the issue. Broke yum when cancelling the installation script partway through so will update when I've had a play with `view-config`...

File: scripts/os/el7.sh [hunk 2]
```diff
@@ -27,6 +27,7 @@ install_runtime_prerequisites() {
         yum -e0 -y install libpcap && \
         yum -e0 -y install ipmitool && \
         yum -e0 -y install gettext && \
+        yum -e0 -y install epel-release \
```
Comment (bobwhitelock) [inline scripts/os/el7.sh]: This isn't going to work is it? Missing the `&&`

**Comment by ColonelPanicks** [comment]
Alright, finally used `view-config` and it's beautiful 😍 

Is this an issue or am I misusing `-i`? (note: `metal configure domain` and `metal configure group nodes` have both been performed before running this)

```
[root@deploy_test ~]# metal view-config -i nodes
error: Unset template parameter: pri_network_defined. Use --trace to view backtrace
```

Trace output
```
[root@deploy_test ~]# metal view-config -i nodes --trace
/opt/metalware/src/templating/missing_parameter_wrapper.rb:36:in `method_missing': Unset template parameter: pri_network_defined (Metalware::MissingParameterError)
	from (erb):1:in `block in replace_erb'
	from /opt/metalware/opt/ruby/lib/ruby/2.4.0/erb.rb:893:in `eval'
	from /opt/metalware/opt/ruby/lib/ruby/2.4.0/erb.rb:893:in `block in result'
	from /opt/metalware/opt/ruby/lib/ruby/2.4.0/erb.rb:894:in `result'
	from /opt/metalware/src/templating/renderer.rb:32:in `render_erb_template'
	from /opt/metalware/src/templating/renderer.rb:12:in `replace_erb'
	from /opt/metalware/src/templating/repo_config_parser.rb:120:in `parse_string_config_value'
	from /opt/metalware/src/templating/repo_config_parser.rb:108:in `parse_config_value'
	from /opt/metalware/src/templating/repo_config_parser.rb:111:in `block in parse_config_value'
	from /opt/metalware/src/templating/repo_config_parser.rb:110:in `each'
	from /opt/metalware/src/templating/repo_config_parser.rb:110:in `map'
	from /opt/metalware/src/templating/repo_config_parser.rb:110:in `parse_config_value'
	from /opt/metalware/src/templating/repo_config_parser.rb:111:in `block in parse_config_value'
	from /opt/metalware/src/templating/repo_config_parser.rb:110:in `each'
	from /opt/metalware/src/templating/repo_config_parser.rb:110:in `map'
	from /opt/metalware/src/templating/repo_config_parser.rb:110:in `parse_config_value'
	from /opt/metalware/src/templating/repo_config_parser.rb:101:in `block in perform_config_parsing_pass'
	from /opt/metalware/src/templating/repo_config_parser.rb:100:in `each'
	from /opt/metalware/src/templating/repo_config_parser.rb:100:in `map'
	from /opt/metalware/src/templating/repo_config_parser.rb:100:in `perform_config_parsing_pass'
	from /opt/metalware/src/templating/repo_config_parser.rb:45:in `parse'
	from /opt/metalware/src/templating/repo_config_parser.rb:25:in `parse_for_node'
	from /opt/metalware/src/commands/view_config.rb:23:in `templating_config_json'
	from /opt/metalware/src/commands/view_config.rb:15:in `run'
	from /opt/metalware/src/command_helpers/base_command.rb:37:in `initialize'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/command.rb:217:in `new'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/command.rb:217:in `call'
	from /opt/metalware/src/commander_extensions.rb:61:in `call'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/command.rb:184:in `run'
	from /opt/metalware/src/commander_extensions.rb:49:in `run'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/runner.rb:464:in `run_active_command'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/runner.rb:66:in `run!'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/delegates.rb:15:in `run!'
	from /opt/metalware/src/cli.rb:52:in `run!'
	from /opt/metalware/src/cli.rb:47:in `run'
	from bin/metal:28:in `<main>'
```

Besides that, the domain and node outputs are good.

**Pull request merged**